contracts/libraries/OFTTransportAdapter.sol

```diff
@@ -0,0 +1,114 @@
+     * @param _oftMessenger OFTAdapter contract to bridge to the other chain. If address is set to zero, OFT bridging will be disabled.
+     * @param _oftDstEid The endpoint ID that OFT will transfer funds to.
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        IERC20 _usdt,
+        IOFT _oftMessenger,
+        uint32 _oftDstEid
+    ) {
+        USDT = _usdt;
+        OFT_MESSENGER = _oftMessenger;
+        DST_EID = _oftDstEid;
+    }
+
+    /**
+     * @notice Returns whether or not the OFT bridge is enabled.
+     * @dev If the IOFT is the zero address, OFT bridging is disabled.
+     */
+    function _isOFTEnabled(address _token) internal view returns (bool) {
+        return address(OFT_MESSENGER) != address(0) && address(USDT) == _token;
+    }
+
+    /**
+     * @notice transfer usdt to the other domain via OFT messaging protocol
+     * @param _to address to receive a trasnfer on the destination chain
+     * @param _amount amount to send
+     */
+    function _transferUsdtViaOFT(address _to, uint256 _amount) internal {
+        bytes32 to = _to.toBytes32();
+
+        SendParam memory sendParam = SendParam(
+            DST_EID,
+            to,
+            /**
+             * _amount, _amount here specify `amountLD` and `minAmountLD`. These 2 have a subtle relationship
+             * OFT "removes dust" on .send, which should not affect USDT transfer because of how OFT works internally with 6-decimal tokens.
+             * Setting them both to `_amount` protects us from dust subtraction on the OFT side. If any dust is subtracted, the later .send should revert.
+             */
+            _amount,
+            _amount,
+            /**
+             * EMPTY_MSG_BYTES, EMPTY_MSG_BYTES, EMPTY_MSG_BYTES here specify `extraOptions`, `composeMsg` and `oftCmd`.
+             * These can be set to empty bytes arrays for the purposes of sending a simple cross-chain transfer.
+             */
+            EMPTY_MSG_BYTES,
+            EMPTY_MSG_BYTES,
+            EMPTY_MSG_BYTES
+        );
+
+        // `false` in the 2nd param here refers to `bool _payInLzToken`. We will pay in native token, so set to `false`
+        MessagingFee memory fee = OFT_MESSENGER.quoteSend(sendParam, false);
+
+        // approve the exact _amount for `OFT_MESSENGER` to spend. Fee will be paid in native token
+        USDT.forceApprove(address(OFT_MESSENGER), _amount);
+
+        // todo: not really sure if we should blindly trust the fee.nativeFee here and just send it away. Should we check it against some sane cap?
+        // todo: parent pool calls `_contractHasSufficientEthBalance` on canonical bridge transfer. Should we do the same?
+        // setting `refundAddress` to `ZERO_ADDRESS` here, because we calculate the fees precicely and we can save gas this way
+        (, OFTReceipt memory oftReceipt) = OFT_MESSENGER.send{ value: fee.nativeFee }(sendParam, fee, ZERO_ADDRESS);
```

Comment:
Regarding your comment: 

> // todo: not really sure if we should blindly trust the fee.nativeFee here and just send it away. Should we check it against some sane cap?

I'm not sure what you mean to check. The `value` is being set to `fee.nativeFee` so the OFT_MESSENGER can't take any more ETH from this contract. Is there something else to check?